src/frameworks/next/index.ts

```diff
@@ -84,77 +95,95 @@ export async function build(dir: string): Promise<BuildResult> {
     throw e;
   });
 
-  try {
-    // Using spawn here, rather than their programatic API because I can't silence it
-    // Failures with Next export are expected, we're just trying to do it if we can
-    execSync(`${CLI_COMMAND} export`, { cwd: dir, stdio: "ignore" });
-  } catch (e) {
-    // continue, failure is expected
+  const reasonsForBackend = [];
+  const { distDir } = await getConfig(dir);
+
+  if (await isUsingMiddleware(dir, false)) {
+    reasonsForBackend.push("middleware");
   }
 
-  let wantsBackend = true;
-  const { distDir } = await getConfig(dir);
-  const exportDetailPath = join(dir, distDir, "export-detail.json");
-  const exportDetailExists = await pathExists(exportDetailPath);
-  const exportDetailBuffer = exportDetailExists ? await readFile(exportDetailPath) : undefined;
-  const exportDetailJson = exportDetailBuffer && JSON.parse(exportDetailBuffer.toString());
-  if (exportDetailJson?.success) {
-    const appPathRoutesManifestPath = join(dir, distDir, "app-path-routes-manifest.json");
-    const appPathRoutesManifestJSON = fileExistsSync(appPathRoutesManifestPath)
-      ? await readFile(appPathRoutesManifestPath).then((it) => JSON.parse(it.toString()))
-      : {};
-    const prerenderManifestJSON = await readFile(
-      join(dir, distDir, "prerender-manifest.json")
-    ).then((it) => JSON.parse(it.toString()));
-    const anyDynamicRouteFallbacks = !!Object.values(
-      prerenderManifestJSON.dynamicRoutes || {}
-    ).find((it: any) => it.fallback !== false);
-    const pagesManifestJSON = await readFile(
-      join(dir, distDir, "server", "pages-manifest.json")
-    ).then((it) => JSON.parse(it.toString()));
-    const prerenderedRoutes = Object.keys(prerenderManifestJSON.routes);
-    const dynamicRoutes = Object.keys(prerenderManifestJSON.dynamicRoutes);
-    const unrenderedPages = [
-      ...Object.keys(pagesManifestJSON),
-      // TODO flush out fully rendered detection with a app directory (Next 13)
-      // we shouldn't go too crazy here yet, as this is currently an expiriment
-      ...Object.values<string>(appPathRoutesManifestJSON),
-    ].filter(
-      (it) =>
-        !(
-          ["/_app", "/", "/_error", "/_document", "/404"].includes(it) ||
-          prerenderedRoutes.includes(it) ||
-          dynamicRoutes.includes(it)
-        )
-    );
-    // TODO log these as a reason why Cloud Functions are needed
-    if (!anyDynamicRouteFallbacks && unrenderedPages.length === 0) {
-      wantsBackend = false;
+  if (await isUsingImageOptimization(join(dir, distDir))) {
+    reasonsForBackend.push(`Image Optimization`);
+  }
+
+  const appPathRoutesManifestPath = join(dir, distDir, APP_PATH_ROUTES_MANIFEST);
+  const appPathRoutesManifestJSON = fileExistsSync(appPathRoutesManifestPath)
+    ? await readJSON(appPathRoutesManifestPath)
+    : {};
+  const usingAppDirectory = Object.keys(appPathRoutesManifestJSON).length > 0;
```

Comment:
mind refactor to isUsingAppDirectory? Maybe we can port some of these checks into that util

Reply:
sure, will do that. I'm about to do the same refactors as mentioned [here](https://github.com/firebase/firebase-tools/pull/5320#discussion_r1046604060)

Reply:
actually let's stick with just that the file exists, like is in the util... it is unstable after all, next could break the contract